### PR TITLE
Mul<f32> for ScalingMode

### PR DESCRIPTION
**Pull request by stepancheg**
Complement to https://github.com/bevyengine/bevy/pull/11022: if `OrthographicProjection.scale` is removed, this can be used instead.

CC @doonv @Davier

### CODE REVIEW

**Comment by doonv** [comment]
This shouldn't be a seperate pull request, it should just be a part of https://github.com/bevyengine/bevy/pull/11022

**Comment by stepancheg** [comment]
Might be useful even if #11022 rejected, so I created a separate PR.

**Comment by doonv** [comment]
Ah fair enough.

**Comment by doonv** [comment]
If `ScalingMode` both handles the amount of scaling and the type (mode) of scaling. Maybe it should be renamed to something simpler like `Scaling`?

**Comment by stepancheg** [comment]
Right. Or perhaps `OrthographicProjectionScaling`, to make it clear it has nothing to do with perspective projection. (But either options will be significant breaking change.)

**Comment by doonv** [comment]
Actually, I think `OrthographicScaling` would be the best name for this. Not too long, not too short.

File: crates/bevy_render/src/camera/projection.rs [hunk 1]
```diff
@@ -215,6 +216,45 @@ pub enum ScalingMode {
     FixedHorizontal(f32),
 }
 
+/// Scale the scaling mode. For example, multiplying by 2 makes everything twice as small.
+impl Mul<f32> for ScalingMode {
+    type Output = ScalingMode;
+
+    fn mul(self, rhs: f32) -> ScalingMode {
```
Comment (doonv) [inline crates/bevy_render/src/camera/projection.rs:223]: I don't think adding the doc-comment to the implementation does anything. So I think it should it should be added to the `mul` method itself, which will allow the user to hover over the `*` and see what the multiplication operator does in this case. I also reworded it a bit in my suggestion.
```suggestion
impl Mul<f32> for ScalingMode {
    type Output = ScalingMode;

    /// Scale the [`ScalingMode`]. Multiplying by `2` makes the viewport twice as large.
    fn mul(self, rhs: f32) -> ScalingMode {
```

Comment (stepancheg) [reply crates/bevy_render/src/camera/projection.rs:223]: > I don't think adding the doc-comment to the implementation does anything

It is shown in generated rustdoc:

image

Comment (stepancheg) [reply crates/bevy_render/src/camera/projection.rs]: Anyway, moved the updated doc to `mul`.

**Review by doonv — APPROVED** [review]
Looks good to me!

**Comment by Davier** [comment]
Do you think we should have `Div` and `DivAssign`?

**Comment by doonv** [comment]
> Do you think we should have `Div` and `DivAssign`?

Yeah, why not.

**Comment by stepancheg** [comment]
Added `Div` and `DivAssign`.

**Review by doonv — APPROVED** [review]

**Review by Davier — APPROVED** [review]

**Pull request merged**